File: AtlasWeb/Core/wwwroot/app/NebulaClient/Services/External/Asos112Services.ts
//$58DBE58A
import { Guid } from "NebulaClient/Mscorlib/Guid";
import { TTObjectContext } from "NebulaClient/StorageManager/InstanceManagement/TTObjectContext";
import { Exception } from "NebulaClient/Mscorlib/Exception";
import { IWebMethodCallback } from "NebulaClient/Utils/IWebMethodCallback";

export namespace XXXXXX112Services {
    export class AsyncBase {
        constructor() {
            this.MessageObjectID = Guid.newGuid().toString();
        }

        public MessageObjectID: string;
        public MedulaProvisionObjectID: string;
    }


    export class EUYSTaniGonderParam extends AsyncBase implements IWebMethodCallback {
        constructor() {
            super();
        }

        public ObjectContext: TTObjectContext;
        public WebMethodCallback(returnValue: any, calledParameters: any[], messageException: Exception): boolean {
            if (returnValue !== null) {
                let result: number = <number>returnValue;
                if (result !== 1) {
                    //string medulaProvisionObjectID = MedulaProvisionObjectID;
                    this.HastaKabulCompletedInternal(result, this.ObjectContext);
                    //MedulaHelper.SaveAndDisposeContext(ObjectContext);
                    return false;
                }
                else return true;
            }
            else return true;
        }
        public HastaKabulCompletedInternal(result: number, _context: TTObjectContext): void {

        }
    }


    export class VakaSonlandirParam extends AsyncBase implements IWebMethodCallback {
        constructor() {
            super();
        }

        public ObjectContext: TTObjectContext;
        public WebMethodCallback(returnValue: any, calledParameters: any[], messageException: Exception): boolean {
            //                if (returnValue != null)
            //                {
            //                   int? result = (int?)returnValue;
            //                    if (result != 1)
            //                    {
            //                        //string medulaProvisionObjectID = MedulaProvisionObjectID;
            //                        HastaKabulCompletedInternal(result, ObjectContext);
            //                        //MedulaHelper.SaveAndDisposeContext(ObjectContext);
            //                        return false;
            //                    }
            //                    else
            return true;
        }
    }


    export class WebMethods {

    }
}

File: ATLASDEV/TTObjectClasses/ExternalWebServices/MedulaWebServiceDefinition/Asos112Services/Asos112Services.cs

using System;
using System.Xml;
using System.Data;
using System.Text;
using System.Drawing;
using System.Reflection;
using System.Collections;
using System.Linq;
using System.ComponentModel;
using System.Collections.Generic;
using System.Runtime.Serialization;
using System.Collections.ObjectModel;
using System.Runtime.InteropServices;

using TTUtils;
using TTObjectClasses;
using TTDataDictionary;
using TTCoreDefinitions;
using TTConnectionManager;
using TTInstanceManagement;
using TTDefinitionManagement;
using TTStorageManager.Security;



using TTStorageManager;
using System.Runtime.Versioning;


namespace TTObjectClasses
{
    public  partial class XXXXXX112Services : TTObject
    {
#region Methods
        [Serializable]
        public class AsyncBase
        {
            [NonSerialized]
            protected TTObjectContext objectcontext;

            private string _medulaProvisionObjectID;

            private string _messageObjectID;

            public string MessageObjectID
            {
                get { return _messageObjectID; }
                set { _messageObjectID = value; }
            }

            public string MedulaProvisionObjectID
            {
                get { return _medulaProvisionObjectID; }
                set { _medulaProvisionObjectID = value; }
            }

            public AsyncBase()
            {
                MessageObjectID = Guid.NewGuid().ToString();
            }
        }

        [Serializable]
        public class EUYSTaniGonderParam : AsyncBase, IWebMethodCallback
        {
            public EUYSTaniGonderParam() { }

            //public EUYSTaniGonderParam(TTObjectClasses.HastaKabulIslemleri.provizyonGirisDVO provizyonGirisDVO, string medulaProvisionObjectID)
            //{
            //    _provizyonGirisDVO = provizyonGirisDVO;
            //    MedulaProvisionObjectID = medulaProvisionObjectID;
            //}
            //public TTObjectClasses.HastaKabulIslemleri.provizyonGirisDVO _provizyonGirisDVO;

            #region IWebMethodCallback Members

            public bool WebMethodCallback(object returnValue, object[] calledParameters, Exception messageException)
            {
                if (returnValue != null)
                {
                   int? result = (int?)returnValue;
                    if (result != 1)
                    {
                        //string medulaProvisionObjectID = MedulaProvisionObjectID;
                        HastaKabulCompletedInternal(result, ObjectContext);
                        //MedulaHelper.SaveAndDisposeContext(ObjectContext);
                        return false;
                    }
                    else
                        return true;
                }
                else
                    return true;

            }

            #endregion

            #region IAttributeInterface Members

            public TTObjectContext ObjectContext
            {
                get
                {
                    if (objectcontext == null)
                        objectcontext = new TTObjectContext(false);
                    return objectcontext;
                }
            }

            public void HastaKabulCompletedInternal(int? result, TTObjectContext _context)
            {
                //if (result)
                //{
                //}
            }



            #endregion
        }
        
        
        [Serializable]
        public class VakaSonlandirParam : AsyncBase, IWebMethodCallback
        {
            public VakaSonlandirParam() { }

            //public EUYSTaniGonderParam(TTObjectClasses.HastaKabulIslemleri.provizyonGirisDVO provizyonGirisDVO, string medulaProvisionObjectID)
            //{
            //    _provizyonGirisDVO = provizyonGirisDVO;
            //    MedulaProvisionObjectID = medulaProvisionObjectID;
            //}
            //public TTObjectClasses.HastaKabulIslemleri.provizyonGirisDVO _provizyonGirisDVO;

            #region IWebMethodCallback Members

            public bool WebMethodCallback(object returnValue, object[] calledParameters, Exception messageException)
            {
//                if (returnValue != null)
//                {
//                   int? result = (int?)returnValue;
//                    if (result != 1)
//                    {
//                        //string medulaProvisionObjectID = MedulaProvisionObjectID;
//                        HastaKabulCompletedInternal(result, ObjectContext);
//                        //MedulaHelper.SaveAndDisposeContext(ObjectContext);
//                        return false;
//                    }
//                    else
                        return true;
//                }
//                else
//                    return true;

            }

            #endregion

            #region IAttributeInterface Members

            public TTObjectContext ObjectContext
            {
                get
                {
                    if (objectcontext == null)
                        objectcontext = new TTObjectContext(false);
                    return objectcontext;
                }
            }

//            public void HastaKabulCompletedInternal(int? result, TTObjectContext _context)
//            {
//                //if (result)
//                //{
//                //}
//            }



            #endregion
        }
        
#endregion Methods

    }
}